### PR TITLE
FAI-635: CF analysis integration tests are not checking that the analysis was SUCCESSFUL

### PR DESCRIPTION
**Pull request by manstis**
See https://issues.redhat.com/browse/FAI-635

Many thanks for submitting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

- [x] You have read the [contributors guide](https://github.com/kiegroup/kogito-runtimes#contributing-to-kogito)
- [x] Pull Request title is properly formatted: `KOGITO-XYZ Subject`
- [x] Pull Request title contains the target branch if not targeting main: `[0.9.x] KOGITO-XYZ Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains link to any dependent or related Pull Request
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>Pull Request</b>  
  Please add comment: <b>Jenkins retest this</b>
 
* <b>Quarkus LTS checks</b>  
  Please add comment: <b>Jenkins run LTS</b>

* <b>Native checks</b>  
  Please add comment: <b>Jenkins run native</b>

</details>

### CODE REVIEW

**Review by jiripetrlik — CHANGES_REQUESTED** [review]
Hello @manstis it seems to me that you are trying to fix much more than only integration tests. I would suggest to move "non-test" changes into a separate PR with proper description like refactoring etc.

**Comment by manstis** [comment]
Hi @jiripetrlik Yes, this is a draft PR on top of #1083. The second commit in this PR is for FAI-635.

I will rebase and move to "Ready for review" when #1083 is merged.

**Comment by jiripetrlik** [comment]
@manstis Understand, thank you for explanation. Makes sense.

**Comment by manstis** [comment]
@r00ta @jiripetrlik This PR is now ready for review.

**Comment by manstis** [comment]
Jenkins retest this

**Review by danielezonca — APPROVED** [review]

**Comment by manstis** [comment]
@r00ta @jiripetrlik This PR is now ready for review.

**Comment by manstis** [comment]
SonarCloud is reporting lack of test coverage of a test class.

**Comment by manstis** [comment]
@r00ta @jiripetrlik This PR is now ready for review.

File: apps-integration-tests/integration-tests-trusty-service/integration-tests-trusty-service-common/src/main/java/org/kie/kogito/it/trusty/AbstractTrustyExplainabilityEnd2EndIT.java [hunk 1]
```diff
@@ -216,14 +224,24 @@ public void doTest() {
...
+            await()
+                    .atLeast(5, SECONDS)
+                    .atMost(30, SECONDS)
+                    .with().pollInterval(5, SECONDS)
+                    .untilAsserted(() -> {
+                        executionIds.forEach(executionId -> {
+                            doCounterfactualRequests(trustyService, accessToken, executionId);
+                            doCounterfactualResponses(trustyService, accessToken, executionId);
                         });
                     });
```
Comment (r00ta) [inline apps-integration-tests/integration-tests-trusty-service/integration-tests-trusty-service-common/src/main/java/org/kie/kogito/it/trusty/AbstractTrustyExplainabilityEnd2EndIT.java]: iiuc this is going to create a new cf request for each executionId until the assertions become true. Is this intended? Is it possible that this is requesting multiple CF the the same execution at the same time?

Comment (r00ta) [reply apps-integration-tests/integration-tests-trusty-service/integration-tests-trusty-service-common/src/main/java/org/kie/kogito/it/trusty/AbstractTrustyExplainabilityEnd2EndIT.java]: I mean, if this is not true at the first shot this is going to be executed again -> new CF request . At least until the assertion is true

Comment (manstis) [reply apps-integration-tests/integration-tests-trusty-service/integration-tests-trusty-service-common/src/main/java/org/kie/kogito/it/trusty/AbstractTrustyExplainabilityEnd2EndIT.java]: Yes, this is true and not entirely desirable. 

I've refactored to isolate each execution and counterfactual explanation in a new commit.

WDYT?

File: apps-integration-tests/integration-tests-trusty-service/integration-tests-trusty-service-common/src/main/java/org/kie/kogito/it/trusty/AbstractTrustyExplainabilityEnd2EndIT.java [hunk 2]
```diff
@@ -216,14 +226,26 @@ public void doTest() {
...
+
+            // Request Counterfactuals for each execution and check responses generated
+            executionIds.forEach(executionId -> {
+                await()
+                        .atLeast(500, MILLISECONDS)
+                        .atMost(1, SECONDS)
+                        .with().pollInterval(500, MILLISECONDS)
+                        .until(doCounterfactualRequests(trustyService, accessToken, executionId));
```
Comment (r00ta) [inline apps-integration-tests/integration-tests-trusty-service/integration-tests-trusty-service-common/src/main/java/org/kie/kogito/it/trusty/AbstractTrustyExplainabilityEnd2EndIT.java:238]: What can happen to requests? Is it possible that the request fail for any reason => we have to make it again? 
I understand for the cf response, but what about the request?

Comment (manstis) [reply apps-integration-tests/integration-tests-trusty-service/integration-tests-trusty-service-common/src/main/java/org/kie/kogito/it/trusty/AbstractTrustyExplainabilityEnd2EndIT.java:238]: If a CF request has not been successful (i.e. the `Callable` continues to return `false` in the time limit) something serious has gone wrong and it's correct for the test to fail. It's plausible a request could fail in a production environment for a recoverable reason.. but for this test failure to request is a failed test.

Comment (r00ta) [reply apps-integration-tests/integration-tests-trusty-service/integration-tests-trusty-service-common/src/main/java/org/kie/kogito/it/trusty/AbstractTrustyExplainabilityEnd2EndIT.java:238]: ok, I was thinking to make just one request and fail if it is not working fine. But it's ok, thanks

**Review by r00ta — APPROVED** [review]

**Comment by manstis** [comment]
Hi @jiripetrlik could you please review.

**Review by jiripetrlik — APPROVED** [review]

**Pull request merged**